tests/csi/cstor/volume/provision_test.go

```diff
@@ -0,0 +1,138 @@
+/*
+Copyright 2019 The OpenEBS Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package volume
+
+import (
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+
+	apis "github.com/openebs/maya/pkg/apis/openebs.io/v1alpha1"
+	pvc "github.com/openebs/maya/pkg/kubernetes/persistentvolumeclaim/v1alpha1"
+	sc "github.com/openebs/maya/pkg/kubernetes/storageclass/v1alpha1"
+	spc "github.com/openebs/maya/pkg/storagepoolclaim/v1alpha1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+var _ = Describe("[cstor] [sparse] TEST VOLUME PROVISIONING", func() {
+	var (
+		err     error
+		pvcName = "cstor-volume-claim"
+	)
+
+	BeforeEach(func() {
+		When(" creating a cstor based volume", func() {
+			By("building a storageclass")
+			scObj, err = sc.NewBuilder().
+				WithGenerateName(scName).
+				WithAnnotations(annotations).
+				WithProvisioner(openebsProvisioner).Build()
+			Expect(err).ShouldNot(HaveOccurred(), "while building storageclass obj for storageclass {%s}", scName)
+
+			By("creating above storageclass")
+			scObj, err = ops.SCClient.Create(scObj)
+			Expect(err).To(BeNil(), "while creating storageclass {%s}", scName)
+
+			By("building a storagepoolclaim")
+			spcObj = spc.NewBuilder().
+				WithGenerateName(spcName).
```

Comment:
`WithGenerateName` func willl change the spc name with some suffix values, this will fail to provision the volume .
please refer to latest changes for cstor BDD's.

Reply:
Also we have to create SPC first then SC.
After SPC is created we can use the spc Name to update the SC annotations for `StoragePoolClaim` value.